### PR TITLE
Specs for account/balances and code polishing

### LINKED ISSUE #1665: Member API v2 needs improvements and better documentation
Refactoring
- [ ] Follow REST API rules

- [x] Endpoints not working
    GET: "/accounts/:currency"

- [x] Should be moved/refactored
  POST: "/order/delete" ->  DELETE "order/:id"
  POST: "/orders/clear" -> ?
  GET: "/orders" should have ability to return all orders without filter
  GET: "/accounts/:currency" should allow currency with both upper/lower case.

- [ ] Cancel of order which belongs to other member should raise an error

- [ ] POST: "/order/delete" ->  DELETE "order/:id" returns state: :wait

- [ ]  Improve exceptions mechanism

Documentation
- [ ] Document all APIv2::Entities (follow APIv2::Entities::Account)

- [ ] Sort classes mount (fore calls order)

- [ ] Custom data types should be moved to Entities

### PR DESCRIPTION
**Pull request by mnaichuk**
closes #1665

### CODE REVIEW

**Review by ysv — CHANGES_REQUESTED** [review]
Spec needs to be restructured and refactored

File: spec/api/v2/account/balances_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  let(:response_body) { { 'currency' => 'eth', 'balance' => '30.5', 'locked' => '0.0' } }
+  before do
+    deposit_btc.accept!
+    deposit_eth.accept!
+    withdraw.submit! && withdraw.accept!
+  end
+  describe 'GET api/v2/account/balances' do
+    it 'returns current user balances' do
+      api_get '/api/v2/account/balances', token: token
```
Comment (ysv) [inline spec/api/v2/account/balances_spec.rb]: move to before

File: spec/api/v2/account/balances_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      expect(response).to be_success
```
Comment (ysv) [inline spec/api/v2/account/balances_spec.rb]: separate `it`

File: spec/api/v2/account/balances_spec.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      result = JSON.parse(response.body)
```
Comment (ysv) [inline spec/api/v2/account/balances_spec.rb:26]: extract into separate method

File: spec/api/v2/account/balances_spec.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        { 'currency' => 'trst', 'balance' => '0.0', 'locked' => '0.0' },
+        { 'currency' => 'usd', 'balance' => '0.0', 'locked' => '0.0' },
+        { 'currency' => 'xrp', 'balance' => '0.0', 'locked' => '0.0' }
+      ]
+    end
+
+    it 'returns only balances of enabled currencies' do
+      Currency.find('xrp').update(enabled: false)
+      api_get '/api/v2/account/balances', token: token
+      expect(response).to be_success
```
Comment (ysv) [inline spec/api/v2/account/balances_spec.rb]: `it { is_expected.to have_http_status 200 }`

File: spec/api/v2/account/balances_spec.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      result = JSON.parse(response.body)
+      expect(result.count).to eq Currency.enabled.count
+    end
+  end
+
+  describe 'GET api/v2/account/balances/:currency' do
+    it 'returns current user balance by currency' do
+      api_get '/api/v2/account/balances/eth', token: token
+      expect(response).to be_success
```
Comment (ysv) [inline spec/api/v2/account/balances_spec.rb]: it { is_expected.to have_http_status 200 }

File: spec/api/v2/public/markets_spec.rb [hunk 6]
```diff
@@ -307,4 +307,157 @@ def response_body
...
+        member: member
+      )
+    end
+
+    let(:market) { :btcusd }
+
+    let!(:ask_trade) { create(:trade, ask: ask, created_at: 2.days.ago) }
+    let!(:bid_trade) { create(:trade, bid: bid, created_at: 1.day.ago) }
+
+    it 'should return all recent trades' do
```
Comment (ysv) [inline spec/api/v2/public/markets_spec.rb]: remove `should` everywhere

File: app/api/v2/account/balances.rb [hunk 7]
```diff
@@ -8,7 +8,6 @@ class Balances < Grape::API
         helpers API::V2::NamedParams
         before { authenticate! }
```
Comment (ysv) [inline app/api/v2/account/balances.rb]: Move this line to `Account::Mount` & `Market::Mount`

**Review by ysv — CHANGES_REQUESTED** [review]
Spec needs to be restructured and refactored

File: app/api/v2/mount.rb [hunk 8]
```diff
@@ -34,13 +33,13 @@ class Mount < Grape::API
...
-      mount Market::Mount => 'market'
-      mount Management::Mount => 'management'
+      mount Public::Mount       => :public
+      mount Account::Mount      => :account
+      mount Market::Mount       => :market
+      mount Management::Mount   => :management
 
       # The documentation is accessible at http://localhost:3000/swagger?url=/api/v2/swagger
-      add_swagger_documentation base_path:   PREFIX,
+      add_swagger_documentation base_path:   API::Mount::PREFIX + '/' + API_VERSION,
```
Comment (ysv) [inline app/api/v2/mount.rb]: Use URL join

**Review by ysv — CHANGES_REQUESTED** [review]

**Review by mod — APPROVED** [review]

**Pull request merged**